src/infrastructure/content/NetworkMessageHandler.java

```diff
@@ -0,0 +1,79 @@
+package infrastructure.content;
+
+import networking.INotificationHandler;
+import org.json.JSONObject;
+import java.util.HashMap;
+
+/*
+ * NetworkMessageHandler class implements the INotificationHandler interface.
+ * this includes one method i.e @onMessageReceived()
+ * this class is used to notify and to send data to the UI module 
+ * as new user joined or as a user exit or as message sent by a user
+ * 
+ * @author Talha Yaseen (111701019)
+ */
+public class NetworkMessageHandler implements INotificationHandler {
+	
+	/*
+	 * Creating instance of ContentCommunicator class to access map handler and image map
+	 */
+	private ContentCommunicator cc = new ContentCommunicator();
+	
+	/*
+	 * Accessing mapHandler hashmap from ContentCommunicator class to use it for sending message to UI
+	 */
+	private HashMap<String, IContentNotificationHandler> mapHandler = cc.getMapHandler();
+	
+	/*
+	 * Accessing mapImage hashmap from ContentCommunicator class to access image corresponding to username
+	 */
+	private HashMap<String, String> mapImage = cc.getMapImage();
+	
+	/*
+	 * This variable will store the meta field of json string sent by the Networking module
+	 */
+	private String type;
+	
+	/*
+	 * This variable will store the username of client
+	 */
+	private String userName;
+	
+	/*
+	 * This variable will store the image accessed by mapImage corresponding to username 
+	 */
+	private String userImage;
+	
+	/*
+	 * Getting handler of the UI module to send the data
+	 */
+	 private IContentNotificationHandler handler;
+	 
+	/*
+	 * @param: json string message (Meta fields are newUser or message or userExit)
+	 * if meta field is newUser, then remaining field would only be username
+	 * if meta field is message, then remaining fields would be username, message and time
+	 * if meta field is exitUser, then remaining field would only be username
+	 * Networking module will be calling this method with json string message to send data to the content module
+	 */
+	 
+	public void onMessageReceived(String message) {
+		JSONObject jsonObject = new JSONObject(message);
+		type = jsonObject.getString("meta");
+		handler = mapHandler.get("UI");
+		jsonObject.remove("meta");
```

Comment:
I didn't understand that you were saving the handler variable in the class. In spite of saving it, you seem to be getting the handler at every call. Have checks to see if it is saved.